packages/now-node-bridge/src/bridge.ts

```diff
@@ -118,27 +118,34 @@ export class Bridge {
 
   setServer(server: Server) {
     this.server = server;
-    server.once('listening', () => {
-      const addr = server.address();
-      if (typeof addr === 'string') {
-        throw new Error(`Unexpected string for \`server.address()\`: ${addr}`);
-      } else if (!addr) {
-        throw new Error('`server.address()` returned `null`');
-      } else {
-        this.resolveListening(addr);
-      }
-    });
   }
 
   listen() {
     if (!this.server) {
       throw new Error('Server has not been set!');
     }
 
-    return this.server.listen({
-      host: '127.0.0.1',
-      port: 0,
-    });
+    const resolveListening = this.resolveListening;
+
+    return this.server.listen(
+      {
+        host: '127.0.0.1',
+        port: 0,
+      },
+      function() {
```

Comment:
You should be able to add a type for `this`. Off the top of my head, I think:

```suggestion
      function(this: Server) {
```

Reply:
Also what is the benefit of `this` vs using the `this.server` above?

Reply:
Isn't that typing the first parameter of the function?

Reply:
@lucleray See the docs here: https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#specifying-the-type-of-this-for-functions

Reply:
`this.server` is a reference to the "duck-typed" server (we duck type the server by looking for a listen method). In some cases, `this.server` is not a real http.Server. For example, `app = express()` is not a real http.Server and does not have the `address()` method.

I'll add a comment for that.

Reply:
Ok, I updated my suggestion above to be an optional address method.